taxonomy-manager-engine/src/main/java/com/digirati/taxonomy/manager/storage/record/mapper/ConceptRelationshipRecordMapper.java

```diff
@@ -0,0 +1,24 @@
+package com.digirati.taxonomy.manager.storage.record.mapper;
+
+import com.digirati.taxman.common.taxonomy.ConceptRelationshipType;
+import com.digirati.taxonomy.manager.storage.record.ConceptRelationshipRecord;
+import org.springframework.jdbc.core.RowMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+
+public class ConceptRelationshipRecordMapper implements RowMapper<ConceptRelationshipRecord> {
+    @Override
+    public ConceptRelationshipRecord mapRow(ResultSet rs, int rowNum) throws SQLException {
+        UUID source = rs.getObject("source_uuid", UUID.class);
+        UUID target = rs.getObject("target_uuid", UUID.class);
+        ConceptRelationshipType type = ConceptRelationshipType.valueOf(rs.getString("type"));
+        boolean transitive = rs.getBoolean("transitive");
+        ConceptRelationshipRecord record = new ConceptRelationshipRecord(source, target, type, transitive);
+
+        record.setPreferredLabel(ResultSetUtils.getPlainLiteralMap(rs, "target_preferred_label"));
```

Comment:
> but I think we'll also need the same for the source
Why would we?

Reply:
> Why would we?

I explained that in the sentences following your quote. To add an example for clarity, if we have a relationship that says concept B is narrower than concept A this will only include concept A's preferred label. When we get concept A, I would expect to get all relationships in which A is broader than the target, or the source is narrower than A. For the relationship described above having concept A's preferred label on the relationship is pointless because we have concept A itself. What we need in this scenario is the source preferred label, otherwise we're going to have to make an additional database call to get it in order to display it in the UI. So this code should be changed to provide both